### PR TITLE
Use query syntax instead of foreach 

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Related issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/2592

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/tests/complex-query-test.bal [hunk 1]
```diff
@@ -292,19 +292,19 @@ function testMultipleRecordRetrieval() returns error? {
...
+    error? e = check from record{} value in streamData
+        do {
+            if value is ResultMap && counter == 0 {
+                mixTypesActual = value;
+            }
+            counter = counter + 1;
+        };
     if e is error {
         test:assertFail("Error when iterating through records " + e.message());
     }
```
Comment (niveathika) [inline ballerina/tests/complex-query-test.bal:304]:
```suggestion
    check from record{} value in streamData
        do {
            if value is ResultMap && counter == 0 {
                mixTypesActual = value;
            }
            counter = counter + 1;
        };
```
Check in other places as well

Comment (kaneeldias) [reply ballerina/tests/complex-query-test.bal]: Fixed in 15e9375

File: ballerina/tests/call-procedures-test.bal [hunk 2]
```diff
@@ -101,22 +98,19 @@ function testCallWithStringTypesInParams() returns error? {
...
-        test:assertFail("Call procedure insert did not work properly");
-    } else {
-        StringDataForCall expectedDataRow = {
-            varchar_type: "test1",
-            charmax_type: "test2     ",
-            char_type: "c",
-            charactermax_type: "test3     ",
-            character_type: "d",
-            nvarcharmax_type: "test4"
+    error? e = check from StringDataForCall data in queryData
```
Comment (daneshk) [inline ballerina/tests/call-procedures-test.bal:101]:
```suggestion
    check from StringDataForCall data in queryData
```

File: ballerina/tests/numerical-query-test.bal [hunk 3]
```diff
@@ -80,9 +81,10 @@ function testQueryNumericTypeRecord() returns error? {
     MockClient dbClient = check new (url = jdbcURL, user = user, password = password);
     stream<NumericTypeForQuery, Error?> streamData = dbClient->query(`SELECT * FROM NumericTypes`);
     NumericTypeForQuery? returnData = ();
-    check streamData.forEach(function(NumericTypeForQuery data) {
-        returnData = data;
-    });
+    _ = check from NumericTypeForQuery data in streamData
```
Comment (daneshk) [inline ballerina/tests/numerical-query-test.bal:84]:
```suggestion
    check from NumericTypeForQuery data in streamData
```

Comment (kaneeldias) [reply ballerina/tests/numerical-query-test.bal]: Fixed in 5eacd80

File: ballerina/tests/call-procedures-test.bal (hunk 2 above)
Comment (kaneeldias) [reply ballerina/tests/call-procedures-test.bal]: Fixed in 5eacd80

**Review by niveathika — APPROVED** [review]

**Pull request merged**